### DISCUSSION TITLE
How to re-use a flyout

### DISCUSSION THREAD

**Discussion by Omsad** [AvaloniaUI/Avalonia#17047] [Q&A] [answered]
Hi,

Is it possible to declare a flyout as a static resource and then link it to multiple controls?

E.g. define the flyout in one file:
```
<ResourceDictionary>
    <WhatDoIPutHere x:Key="MySharedFlyout">
        <MyFlyout>
            <MyFlyoutItem Text="Blah...">
        </MyFlyout>
    </WhatDoIPutHere>
</ResourceDictionary>
```

And then in other files, across multiple types of controls:
```
<Button FlyoutBase.AttachedFlyout={StaticResource MySharedFlyout} />
<TextBox FlyoutBase.AttachedFlyout="{StaticResource MySharedFlyout}" />
```

**Comment by maxkatz6** [comment, marked as answer]
```xaml
<ResourceDictionary>
        <MyFlyout x:Key="MySharedFlyout">
            <MyFlyoutItem Text="Blah...">
        </MyFlyout>
</ResourceDictionary>
```

Should be enough. Flyout class is not visual, it can be used with multiple parents.

**Reply by Omsad** [reply]
Thank you, I assumed I was missing something obvious but couldn't find any documentation about sharing a flyout.

**Reply by Sewer56** [reply]
Note that if you do this, and there's commands etc. attached via binding, it will only ever assign the command once if inherited from DataContext.

On subsequent reuses, old commands will be used. That may be problematic depending on context. Not yet sure of cleanest workaround (may just be `Template` the inside of the flyout), looking into it; but not feeling very positive about it so far.

**Reply by Omsad** [reply]
I wanted to re-use the flyout in a datagird and so bound the commands through the parent column header and a converter, e.g. so a command could be returned per column header and key.

All these commands essentially resolved to the same set of commands on the datagrid, so this hasn't been an issue for me. I do however have a vague memory of having issues binding commands, so maybe this is why I did it this way.
